dev/packages/alpha/kafka/0.0.2/manifest.yml

```diff
@@ -0,0 +1,77 @@
+format_version: 1.0.0
+name: kafka
+title: Kafka
+version: 0.0.2
+license: basic
+description: Kafka Integration
+type: integration
+categories:
+- logs
+- metrics
+release: beta
+removable: true
+requirement:
+  kibana:
+    versions: '>=7.3.0 <8.0.0'
+  elasticsearch:
+    versions: '>7.0.1'
+screenshots:
+- src: /img/filebeat-kafka-logs-overview.png
+  title: filebeat kafka logs overview
+  size: 1363x657
+  type: image/png
+- src: /img/metricbeat_kafka_dashboard.png
+  title: metricbeat kafka dashboard
+  size: 1726x711
+  type: image/png
+icons:
+- src: /img/logo_kafka.svg
+  title: logo kafka
+  size: 32x32
+  type: image/svg+xml
+datasources:
+- name: kafka
+  title: Kafka logs and metrics
+  description: Collect logs and metrics from Kafka instances
+  inputs:
+  - type: logs
+    title: Collect logs from Kafka instances
+    description: Collecting Kafka log logs
+  - type: kafka/metrics
+    title: Collect metrics from Kafka instances
+    description: Collecting Kafka broker, consumer, consumergroup, partition and producer
+      metrics
+    vars:
+      - name: hosts
+        type: text
+        title: Kafka address
+        multi: true
+        required: true
+        show_user: true
+        default:
+          - 'localhost:9092'
+      - name: jolokia_hosts
+        type: text
+        title: Address of Jolokia agents installed in Kafka (required for some metricsets)
+        multi: true
+        required: false
+        show_user: true
+        default:
+          - 'localhost:8778'
```

Comment:
This module has some metricsets that need to connect to Kafka, and some others that need to connect to Jolokia. I am defining two "hosts" variables, one for each kind of metricset.

Reply:
Are there any metricsets that need to connect to both? If not, I think we should specify two different inputs instead.

Reply:
Oh, right, makes sense, I didn't think on this possibility, I will give a try, thanks!

Reply:
How can I define two different inputs for the same module? I have tried with `input: kafka_jolokia/metrics`, but then it configures metricbeat with `module: kafka_jolokia`, that doesn't exist. And with `input: kafka/jolokia_metrics` it doesn't create any configuration.

Reply:
I think you should leave the main manifest.yml as is (2 inputs: logs and kafka/metrics) and go deeper to datasets. `agent/stream/stream.yml.hbs` is the configuration of metricbeat that is delivered to the agent. If you want to take a look at integration with 2 stream configurations, you can take a look at "o365".

Reply:
I think the problem here is that on the Metricbeat side we have 1 module using 2 different inputs. We will face the same issue in Kubernetes. Until we have support for inputs in Metricbeat I think we must add some logic to the agent to do the correct routing.

@ph @michalpristas In summary, the Metricbeat Kafka module uses two different inputs: One is Kafka, the other one is Jolokia but both of the code is inside the Kafka module. The config file here makes it more obvious: https://github.com/elastic/beats/blob/master/metricbeat/module/kafka/_meta/config.yml (see the different hosts configs). We have the same challenge with Kubernetes: https://github.com/elastic/beats/blob/master/metricbeat/module/kubernetes/_meta/config.yml

@jsoriano To unblock you what I suggest for now is to just have all input configs inside the dataset instead of the package manifest. It means the user has to change the host info in multiple places but at least we have it working until Agent supports multiple inputs per module. An other workaround could be to support named inputs. So the same input can be specified multiple times on the package level but a name is given, so the dataset will match the correct one 🤔

Reply:
> To unblock you what I suggest for now is to just have all input configs inside the dataset instead of the package manifest. It means the user has to change the host info in multiple places but at least we have it working until Agent supports multiple inputs per module.

Well, my previous option of having two hosts with a single metrics input in the package manifest was also working, I could revert to this one. The good thing of this option is that hosts have to be configured in a single place, the bad thing is that it is not obvious what stream uses each host, but it could be added as part of the description.

Reply:
I forgot that we can two different names for the config even though it is called the same in the end. Nice feature :-) +1 on what you had initially for now.

When you think of monitoring Kafka, you think of it of monitoring two different "inputs / parts" or it just one thing you want to configure in one input?

Reply:
> When you think of monitoring Kafka, you think of it of monitoring two different "inputs / parts" or it just one thing you want to configure in one input?

Yeah, I am thinking that maybe we could separate this into multiple integrations, we have now these metricsets:
* `partition` collects metrics from the kafka broker, using kafka protocol
* `consumergroup` collects metrics from the kafka broker, using kafka protocol
* `broker` collects metrics from the kafka broker, using jolokia
* `consumer` collects metrics from kafka consumers (implemented in java), using jolokia
* `producer` collects metrics from kafka consumers (implemented in java), using jolokia

I think that most of the cases they are going to be used separatedly:
* `partition`, `consumergroup` and optionally `broker` to monitor brokers
* `consumer` to monitor consumers
* `producer` to monitor producers

Would it make sense to separate this into three integrations? I think it could be more clear. We would need in any case a separated setting for the jolokia host of `broker`, but it could be defined only in this dataset.

Reply:
I have pushed the change for the approach with three integrations (not very tested yet).

Reply:
Sorry for the late answer. When you say 3 integrations, you mean 3 inputs? Could you share a screenshot on how it looks now?

Reply:
Sorry, I think I mean three packages. Screenshots are updated.

Reply:
Umm, is it possible to have three different datasources in the same package? Would it work the same as the three packages I am adding now?

Reply:
We had a discussion offline about this, we have decided to go on by now only with the metricsets intended to monitor brokers, so I will remove the consumer and producer metricsets from this PR.